docs/source/model_design/programming_vivarium/artifact_building/index.rst

```diff
@@ -29,12 +29,156 @@ Artifact Building
+more than one data key. Check if you want to change how data is pulled for
+all data keys using that function, or only one, and edit accordingly.
+
+If you're being asked to add a new data key, or write an artifact from scratch,
+a good place to start is from the `vivarium research template <https://github.com/ihmeuw/vivarium_research_template>`_. In this repo's version of :code:`loader.py`
+you can find some basic examples of creating the loader functions.
+
+Most of the time, data is loaded using the :code:`load_standard_data` function,
+which pulls data from GBD into Vivarium formatting. However, there might be
+alternative functions for population data keys, or if you want data in non-standard
+formatting.
+
+Additionally, consider if you need all of the data to be saved. For
+example, in the child nutrition model, we don't use data for age
+groups over 5 years old, so we filter out some of the pulled data to
+limit the size of the artifact.
+
+For data keys that don't pull information from GBD, the process of writing
+or editing these functions is largely the same. Instead of :code:`load_standard_data`
+you would generally just use :code:`read_csv` from a specified data path. The
+path to each data input is stored in the :code:`paths.py` file. If you update a
+data input, update the reference in the :code:`paths.py` file to ensure you're
+pulling the most up to date information.
+
+Other Input Data
+----------------
+
+Data keys and their functions make up the bulk of the data pulling
+infrastructure for artifact building. However, there might be some other
+metadata that's important to be aware of. Most of this can be found
+in the :code:`metadata.py` file.
+
+This is information like draw counts, index columns, and locations that
+can be run. If you're starting a new model, you'll have to write
+this information, but it should largely match what is present in
+the `vivarium research template <https://github.com/ihmeuw/vivarium_research_template>`_.
+
+If you are editing an existing model, just check to make sure this
+information matches what you expect.
+
+As referenced above, there is also a :code:`paths.py` file which
+cotains the file paths and names for all other input data.
+
+Building an Artifact on the Command Line
+----------------------------------------
+
+Before you try to build your artifact, make sure the data inputs are
+up to date. Consider what you are changing in the new model version
+and if that will impact the artifact data, and make any needed updates
+to the files and functions, as outlined above.
+
+Creating an Environment
++++++++++++++++++++++++
+
+First, you will need to create an environment. This will be the same
+as creating usual environments, except when you install the project repo,
+you'll need to do a special editable install. The command looks like this :code:`pip install -e.[data]`. Some other packages can be helpful here as well like :code:`ipython` for
```

Comment:
I'd suggest being even more explicit that this will need to be a different environment from that which you use to run models, and that the model for running the model should use `.[dev]` whereas the one for building the artifact should be `.[data]`.